### PR TITLE
Test plumbing, a few initial example tests, a few bug fixes

### PR DESCRIPTION
**Pull request by aviflax**

### CODE REVIEW

File: test/examples/md2c8e/links_test.clj [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+(deftest test-page?
+  (are [expected given] (= expected (#'links/page? given))
+    false nil
+    false {}
+    false ""
+    false {:page-id 0}
+    false {::c8e/id 0}
+    true  {::c8e/page-id 0}))
+
+(defn- page
```
Comment (mattsoutherden) [inline test/examples/md2c8e/links_test.clj:18]: Can this share the helper with core-test?

Comment (aviflax) [reply test/examples/md2c8e/links_test.clj:18]: Absolutely could, but it doesn’t meet the Rule of Three, and also I tend to relax DRY in test code, so I usually apply the “Rule of Four” in that context. 😬

Comment (mattsoutherden) [reply test/examples/md2c8e/links_test.clj:18]: IIUC, The Rule of Three is about identifying patterns with more certainty before extracting. In this case, I think the pattern - build the data/metadata structure from `file->page`  - was probably immediately identifiable to you when you first wanted to create the helper.

However, not a blocker.

Comment (aviflax) [reply test/examples/md2c8e/links_test.clj:18]: 🤯 🤦  of course! How am I so dumb? Will fix. Thank you!

**Pull request merged**